.circleci/config.yml

```diff
@@ -0,0 +1,653 @@
+        default: ''
+      skip_checkstyle:
+        type: boolean
+        default: true
+      skip_tests:
+        type: boolean
+        default: true
+      m2_repo:
+        type: string
+        default: *m2_repo_path
+      restore_m2_cache:
+        type: boolean
+        default: false
+      save_m2_cache:
+        type: boolean
+        default: false
+      background:
+        type: boolean
+        default: false
+      root_path:
+        type: string
+        default: *root_path
+    steps:
+      - when:
+          condition: << parameters.restore_m2_cache >>
+          steps:
+            - restore_cache: *m2_cache_keys
+      - run:
+          name: 'Run maven <<parameters.phase>> with options: <<parameters.additional_options>>'
+          command: |
+            CHECKSTYLE_OPTION=''
+            if [[ << parameters.skip_checkstyle >> = true ]]; then
+              CHECKSTYLE_OPTION='-Dcheckstyle.skip'
+            fi
+            SKIPTESTS_OPTION='-DskipTests=<< parameters.skip_tests >>'
+            mvn <<parameters.phase>> --batch-mode \
+                    -f <<parameters.pom_file>> \
+                    $CHECKSTYLE_OPTION \
+                    $SKIPTESTS_OPTION \
+                    -Ditext.license=output-config/itextkey.xml \
+                    -Dddp.firecloudKeysDir=output-config/fc_keys \
+                    -Dconfig.file=output-config/testing-circleci.conf  \
+                    -Dmaven.repo.local=<< parameters.m2_repo >> <<parameters.additional_options>>
+          background: <<parameters.background>>
+      - when:
+          condition: << parameters.save_m2_cache >>
+          steps:
+            - save_cache:
+                key: *m2_cache_key
+                paths:
+                  - << parameters.root_path >>/.m2
+      - store_test_results:
+          path: target/surefire-reports
+
+  wait_for_server:
+    description: Wait for a server to respond for specific period of time
+    parameters:
+      server_address:
+        type: string
+        default: '127.0.0.1'
+      port:
+        type: integer
+      timeout:
+        type: integer
+        default: 60
+    steps:
+      - run:
+          name: Waiting for server to start
+          command: |
+            for i in `seq 1 << parameters.timeout >>`;
+            do
+              nc -z << parameters.server_address >> << parameters.port >> && echo Success && exit 0
+              echo -n .
+              sleep 1
+            done
+            echo Failed waiting for << parameters.server_address >> << parameters.port >> to start && exit 1
+
+  start-ddp-test-server:
+    description: Run test server that applies liquibase and load test data
+    steps:
+      - run-maven-with-pepper-options:
+          phase: exec:java
+          additional_options: '-Dexec.mainClass=org.broadinstitute.ddp.route.IntegrationTestSuite -Dexec.classpathScope=test'
+          background: true
+      - wait_for_server:
+          port: 5556
+          timeout: 60
+
```

Comment:
How we start a DataDonationPlatform server that will be used by (most) of our tests.